extension/src/debugger/debuggerExtensions.ts

```diff
@@ -6,13 +6,14 @@ import { extensionLogOutputChannel } from "../utils/logging";
 import { projectDebuggerExtension } from "./languages/dotnet";
 import { isCsharpInstalled, isPythonInstalled } from "../capabilities";
 import { pythonDebuggerExtension } from "./languages/python";
+import { isDirectory } from "../utils/io";
 
 // Represents a resource-specific debugger extension for when the default session configuration is not sufficient to launch the resource.
 export interface ResourceDebuggerExtension {
     resourceType: string;
     debugAdapter: string;
     extensionId: string | null;
-    displayName: string;
+    getDisplayName: (launchConfig: ExecutableLaunchConfiguration) => string;
```

Comment:
This will allow us to use additional launch configuration context in calculating a display name, ie we could note that the configuration is a module